### PR TITLE
Copy @interactors/html package from bigtest repository

### PR DESCRIPTION
**Pull request by cherewaty**
## Motivation
https://github.com/thefrontside/bigtest/issues/981

## Approach
https://docs.github.com/en/get-started/using-git/splitting-a-subfolder-out-into-a-new-repository

I've used `git-subtree` for this type of chore before, but `git-filter-repo` is waaaaaaay faster to run.

All git history for `packages/interactor` in the `bigtest` repo should carry over here.

## Next steps
- Copy over the CI and tooling configs needed for this package from the `bigtest` repo root. I'm not planning to try and preserve git history for those, since their history will still be maintained in the other repo.
- Turn on branch protections here.
- Delete `packages/interactor` from the `bigtest` repo.

### CODE REVIEW

File: packages/html/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Interaction } from './interaction';
+import { Page } from './page';
```
Comment (cowboyd) [inline packages/html/src/app.ts:2]: We should remember to remove deprecated apis

File: packages/html/src/converge.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { performance } from '@bigtest/performance';
+import { bigtestGlobals } from '@bigtest/globals';
```
Comment (cowboyd) [inline packages/html/src/converge.ts:2]: We should record work to extract interaxtor globals from bigtest globales and to use that opportunity to make them more ergonomic for things like cypress integration

File: packages/html/package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "dtslint": "^4.0.5",
+    "expect": "^24.9.0",
+    "express": "^4.17.1",
+    "jsdom": "^16.2.2",
+    "mocha": "^6.2.2",
+    "ts-node": "^9.1.1"
+  },
+  "volta": {
+    "node": "12.16.0",
+    "yarn": "1.19.1"
```
Comment (cowboyd) [inline packages/html/package.json:49]: We should move this repo to use npm

File: packages/html/src/matcher.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  description?(): never;
+}
+
+export type MaybeMatcher<T> = Matcher<T> | T;
+
+export function isMatcher<T>(value: MaybeMatcher<T>): value is NewMatcher<T> {
+  return value && typeof (value as NewMatcher<T>).match === 'function' && typeof (value as NewMatcher<T>).description === 'function';
+}
+
+export function isDeprecatedMatcher<T>(value: MaybeMatcher<T>): value is OldMatcher<T> {
```
Comment (cowboyd) [inline packages/html/src/matcher.ts:23]: We need to record the work to remove deprecated marchers and also the applyMatcher api

**Review by cowboyd — APPROVED** [review]

**Comment by cowboyd** [comment]
Is publishing set up to work on this repository?

**Comment by cherewaty** [comment]
>Is publishing set up to work on this repository?

PR incoming.

File: packages/html/src/converge.ts (hunk 2 above)
Comment (cherewaty) [reply packages/html/src/converge.ts:2]: I'll file an issue for this.

Comment (cherewaty) [reply packages/html/src/converge.ts:2]: https://github.com/thefrontside/interactors/issues/3

File: packages/html/package.json (hunk 3 above)
Comment (cherewaty) [reply packages/html/package.json:49]: https://github.com/thefrontside/interactors/issues/4

File: packages/html/src/matcher.ts (hunk 4 above)
Comment (cherewaty) [reply packages/html/src/matcher.ts:23]: https://github.com/thefrontside/interactors/issues/5

**Pull request merged**